### PR TITLE
Jw/sc 3029/high fi cards for new timeline events

### PR DESCRIPTION
**Pull request by wittejm**
Screen Shot 2022-03-17 at 10 46 17 AM

### CODE REVIEW

File: src/components/Inbox/MeetingCard.tsx [hunk 1]
```diff
@@ -1,29 +1,67 @@
...
+            {`${
+              meeting.meetingType === MeetingType.VoiceCall &&
+              meeting.status === MeetingStatus.NoShow
+                ? "Missed "
+                : ""
+            } ${getVisitLabel(meeting.meetingType)} •
+            ${
+              meeting.meetingType === MeetingType.VoiceCall
+                ? millisecondsToHoursOrMinutesLabel(meeting.call?.duration)
+                : !!meeting.interval.seconds
```
Comment (wittejm) [inline src/components/Inbox/MeetingCard.tsx:56]: admittedly, some of this logic gets a little tortured with the number of different cases. Voice call duration is the interesting value here, but for scheduled meetings it's nice to see the round numbers seen in `interval`

Comment (cainlevy) [reply src/components/Inbox/MeetingCard.tsx:56]: yeah it's starting to look like voice calls could have their own card

File: src/graphql/queries/GetConnectionTimeline.graphql [hunk 2]
```diff
@@ -26,6 +26,11 @@ query GetConnectionTimeline($id: ID!) {
             }
             ... on Meeting {
               ...MeetingFragment
+              call {
+                ... on Call {
+                  ...CallFragment
+                }
```
Comment (cainlevy) [inline src/graphql/queries/GetConnectionTimeline.graphql:32]: shouldn't need the `... on Call` here.

i'm a little conflicted here. it's nice to use the fragment, but we only need duration, but i could probably add a query optimization to the backend so the other fields are less expensive ... yeah i'll just optimize the resolver.

File: src/utils/Time.ts [hunk 3]
```diff
@@ -27,6 +27,16 @@ export function intervalSecondsToMinutesLabel(seconds?: number | null): string {
...
 }
 
+// Like the above, but round hour amounts are rendered as such.
+export function millisecondsToHoursOrMinutesLabel(
+  milliseconds?: number | null
+): string {
+  const mins = milliseconds && Math.round(milliseconds / (60 * 1000));
+  const hours = mins && mins % 60 === 0 ? mins / 60 : null;
+  if (hours) return `${hours} ${hours === 1 ? "hour" : "hrs"}`;
+  return `${mins} ${mins === 1 ? "min" : "mins"}`;
```
Comment (cainlevy) [inline src/utils/Time.ts:37]: does this need to translate? hmm, that could be existing tech debt in our time utils.

**Review by cainlevy — APPROVED** [review]

File: src/components/Inbox/MeetingCard.tsx (hunk 1 above)
Comment (wittejm) [reply src/components/Inbox/MeetingCard.tsx:56]: Ah yea that will help

File: src/graphql/queries/GetConnectionTimeline.graphql (hunk 2 above)
Comment (wittejm) [reply src/graphql/queries/GetConnectionTimeline.graphql:32]: ^^ fragments are supposed to be cheeeap!

**Pull request merged**